[Timothy Tay] iP

tim0tay opened the pull request:
## Prophet
> **1** The Spirit of the Lord God is upon me; because the Lord hath anointed me to preach good tidings unto the meek; he hath sent me to bind up the brokenhearted, to proclaim liberty to the captives, and the opening of the prison to them that are bound; **2** To proclaim the acceptable year of the Lord, and the day of vengeance of our God; to comfort all that mourn - the prophet Isaiah

Maybe this will be able to do that one day! But for now: 
Prophet frees your mind of having to remember things you need to do. It's:
- text-based
- easy to understand
- ~FAST~ SUPER FAST to use

All you need to do is: 
1. download it from [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ&pp=ygUIcmlja3JvbGw%3D)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **free**!

Features: 
- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Managing events
- [ ] Reminders (coming soon)
---

If you are a Java programmer, you can use this to practice too. Here's the `main` method: 
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

tim0tay committed:
Level-0: Rename, Greet, Exit

tim0tay committed:
Level-1: Echo

tim0tay committed:
Level 2: Add, List

tim0tay committed:
Level 3: Mark as Done

tim0tay committed:
Level 4: ToDos, Events, Deadlines

tim0tay committed:
A-TextUiTesting

tim0tay committed:
Level 5: Handle Errors

tim0tay committed:
Level 6: Delete

awc1116 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,27 @@
+public class Deadline extends Task {
+    protected static final String type = "[D]";
```
Would it be better to make this private instead of protected since it is not used by any other classes in the same package?

awc1116 commented on src/main/java/Prophet.java:
```diff
@@ -0,0 +1,177 @@
+    public Prophet() {
+        this.tasks = new TodoList();
+    }
+    public void inputInterpretator(String str) throws UnrecognisedCommandException {
```
Would it be cleaner to create a separate class of inputInterpretator so that you do not cluster your main class?

awc1116 commented on src/main/java/Prophet.java:
```diff
@@ -0,0 +1,177 @@
+    /**
+     * Prints a string that lists all tasks.
+     */
+    public void printList() {
```
Same for all these methods. Would it be easier to create separate classes for the different techniques call so that when you want to add, edit, or delete certain commands or methods it would be easier and cleaner?

tim0tay committed:
branch-Level-7

tim0tay committed:
Merge branch 'branch-Level-7'

tim0tay committed:
branch-Level-8

tim0tay committed:
Merge branch 'branch-Level-8'

tim0tay committed:
A-MoreOOP

tim0tay committed:
Merge branch 'A-MoreOOP'

tim0tay committed:
A-Packages

soonami69 commented on src/main/java/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,35 @@
+    /**
+     * The AddDeadlineCommand class represents the command to add a deadline task.
+     */
+    private static final String horizontalLine = "-------------------------------------------" +
```
I like how you broke this long line so as to not make the line too long!

soonami69 commented on src/main/java/AddToDoCommand.java:
```diff
@@ -0,0 +1,30 @@
+    @Override
+    public void execute(Ui ui, Storage storage) {
+        ToDoTask newTask = new ToDoTask(this.taskDescription);
+        String added = storage.addToList(newTask);
```
Perhaps a more descriptive name instead of `added` would be better?

soonami69 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,192 @@
+    /**
+     * The Parser class takes in user input and interprets it to perform the necessary actions.
+     */
+    private static final String horizontalLine = "--------------------------------------------------------------------------------\n";
```
I think this line may be too long. Consider breaking it into two?

soonami69 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,192 @@
+            return commands;
+        case "mark":
+            try {
+                String[] mark = str.split("mark ", 2);
```
Perhaps a more intuitive variable name here? I don't quite understand what `mark`'s purpose is based on its name.

soonami69 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,192 @@
+                LocalDate to = LocalDate.parse(timeline[1].trim(), DateTimeFormatter.ofPattern("MMM dd yyyy"));
+                commands.add(new AddEventCommand(
+                        CommandType.EVENT, remainingParts[0].trim(), from, to));
+                if (isDone) {
```
I like how you named the boolean variables intuitively!

soonami69 commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,61 @@
+    /**
+     * Prints the farewell message of the chatbot.
+     */
+    public static void bye() {
```
I think this should be a verb. Maybe `sayGoodbye`?

soonami69 reviewed (commented):
Nice, readable code, with good adherence to code style and quality guidelines 👍

tim0tay committed:
Merge branch 'refs/heads/A-Packages'

tim0tay committed:
Merge remote-tracking branch 'origin/add-gradle-support'

tim0tay committed:
A-JUnit

tim0tay committed:
Merge branch 'A-JUnit'

tim0tay committed:
A-JavaDoc

tim0tay committed:
Merge branch 'A-JavaDoc'

tim0tay committed:
A-CodingStandard

tim0tay committed:
Merge branch 'A-CodingStandard'

tim0tay committed:
Level-9: Find

tim0tay committed:
branch-Level-9

tim0tay committed:
Merge branch 'branch-Level-9'

tim0tay committed:
Level-10: GUI

tim0tay committed:
Merge branch 'Level-10'

tim0tay committed:
A-Checkstyle

tim0tay committed:
Merge branch 'A-Checkstyle'

tim0tay committed:
Add assertions where assumptions are made

Some areas of code rely on assumptions.
These assumptions must be checked to prevent errors.

Let's add some assertions in these areas.

tim0tay committed:
Improve code quality according to SE-Edu standards

Certain methods are long and have layers of nesting.
Others have multiple layers of abstraction, which can be
confusing to read.

Let's:
* SLAP hard
* Make the happy path prominent
* Delete dead code
* Minimise code duplication
* Make code obvious
* Avoid long methods
* Avoid deep nesting

tim0tay committed:
Merge pull request #2 from tim0tay/branch-A-Assertions

Add assertions where assumptions are made

tim0tay committed:
Merge branch 'master' into branch-A-CodeQuality

Fix merge conflict with updated master after PR approval

tim0tay committed:
Fix more merge conflicts

tim0tay committed:
Merge pull request #3 from tim0tay/branch-A-CodeQuality

Improve code quality according to SE-Edu standards

tim0tay committed:
branch-A-Streams

tim0tay committed:
branch-B-ViewSchedules

tim0tay committed:
Remove redundant code

tim0tay committed:
Parser.java: tidy up javadoc comments

tim0tay committed:
Parser.java: tidy up javadoc comments

tim0tay committed:
Merge branch 'master' of https://github.com/tim0tay/ip

tim0tay committed:
Add better support for viewing schedule for a particular day

Deadline and event tasks support date-only deadlines,
and do not include specific times of the day.

Schedule viewing hence does not allow for seeing any
particular sequence of events in a day.

Let's:
* include timings for deadline and event tasks
* sort events of the day by the time they are due

tim0tay committed:
Tidy up code and error handling

tim0tay committed:
Update user guide

tim0tay committed:
Update and add new JUnit tests

tim0tay committed:
Update user guide

tim0tay committed:
Update user guide picture source link

tim0tay committed:
Add title bar for application window

tim0tay committed:
Update README.md

tim0tay committed:
Fix checkstyle issues

tim0tay committed:
Update user guide photo link source

tim0tay committed:
Update user guide format

tim0tay committed:
Update user guide format
